src/utils.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utility functions for the Kratos charm."""
+
+from typing import Dict
+from urllib.parse import urlparse, urlunparse
+
+
+def dict_to_action_output(d: Dict) -> Dict:
+    """Convert all keys in a dict to the format of a juju action output."""
+    ret = {}
+    for k, v in d.items():
+        k = k.replace("_", "-")
+        if isinstance(v, dict):
+            v = dict_to_action_output(v)
+        ret[k] = v
+    return ret
+
+
+def normalise_url(url: str) -> str:
+    """Convert a URL to a more userfriendly format.
```

Comment:
Can you please add an example in the docstring to make it clearer what this method does?

Reply:
Done